csrc/scheduler/matmul.cpp

```diff
@@ -72,6 +72,24 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
       outputs[0]->isA<TensorView>(),
       "fusion's output is not an instance of TensorView class");
 
+  {
+    const auto mma_ops = ir_utils::getMmaOps(fusion);
+    TORCH_INTERNAL_ASSERT(
+        mma_ops.size() == 1,
+        "scheduleMatmul supports fusion with single mma op in definition, got ",
+        mma_ops.size());
+    TORCH_INTERNAL_ASSERT(
+        mma_ops.front()->inputLayout().has_value(),
+        "fusion mma op has undefined input layout");
+    TORCH_INTERNAL_ASSERT(
+        mma_ops.front()->inputLayout().value() == params.layout,
```

Comment:
This might be a good idea.

I can add helper function `ir_util::getMmaOpInputLayout(Fusion* fusion)` to have a single interface to access layout information for `MmaOp` in fusion and put it where `params.layout` is currently in use (2 places). Does it make sense?

Reply:
Hmm, what is the benefit of storing this data twice in different places?

Reply:
For now, I think I am fine to leave this as is, I don't think it too much sense on polishing this part too much because  I believe there will be a major refactor on how input layouts are stored and handled when we add NN support. 

Looks like on Volta, mma op can has TN NT NN TT variants:
https://github.com/NVIDIA/cutlass/blob/main/include/cutlass/arch/mma_sm70.h
starting from Turing, mma are just TN and transposes are handled at smem->register load:
https://github.com/NVIDIA/cutlass/blob/main/include/cutlass/arch/mma_sm75.h
but on hopper, wgmma has TN NT NN TT variants again:
https://github.com/NVIDIA/cutlass/blob/main/include/cute/arch/mma_sm90_gmma.hpp

We would need to revisit our design to support all these cases.

Reply:
Sorry, didn't see your reply before posting my new comment, but yeah, whichever you think is better, you can decide.